src/session/client-certificate.c

```diff
@@ -41,6 +43,19 @@
 
 #define CLIENT_CERTIFICATE_DIRECTORY   "/run/cockpit/tls/clients"
 
+static int
+open_proc_pid (pid_t pid)
+{
+    char path[100];
+    int r = snprintf (path, sizeof (path), "/proc/%d", pid);
+    if (r < 0 || r >= sizeof (path))
```

Comment:
This is definitely an assert...

Reply:
No objection, but I must ask: I thought `assert()` was frowned upon as it may get compiled out in production code?

I mean, I don't see how it could fail (/proc plus maximum length of pid_t is well < 100), but I thought I'd take no chances in cockpit-session :grin: